services/worker/tests/job_runners/config/test_parquet_metadata.py

```diff
@@ -0,0 +1,198 @@
+                        dataset="ok",
+                        config="config_1",
+                        split="train",
+                        url="url2",
+                        filename="filename2",
+                        size=0,
+                        num_rows=3,
+                        parquet_metadata_subpath="ok/--/config_1/filename2",
+                    ),
+                ]
+            ),
+            False,
+        ),
+        (
+            "status_error",
+            "config_1",
+            HTTPStatus.NOT_FOUND,
+            {"error": "error"},
+            CachedArtifactError.__name__,
+            None,
+            True,
+        ),
+        (
+            "format_error",
+            "config_1",
+            HTTPStatus.OK,
+            {"not_parquet_files": "wrong_format"},
+            PreviousStepFormatError.__name__,
+            None,
+            True,
+        ),
+    ],
+)
+def test_compute(
+    app_config: AppConfig,
+    get_job_runner: GetJobRunner,
+    dataset: str,
+    config: str,
+    upstream_status: HTTPStatus,
+    upstream_content: Any,
+    expected_error_code: str,
+    expected_content: Any,
+    should_raise: bool,
+) -> None:
+    upsert_response(
+        kind="config-parquet",
+        dataset=dataset,
+        config=config,
+        content=upstream_content,
+        http_status=upstream_status,
+    )
+    job_runner = get_job_runner(dataset, config, app_config)
+    if should_raise:
+        with pytest.raises(Exception) as e:
+            job_runner.compute()
+        assert e.type.__name__ == expected_error_code
+    else:
+        with patch("worker.job_runners.config.parquet_metadata.ParquetFile") as mock_ParquetFile:
+            mock_ParquetFile.return_value = pq.ParquetFile(dummy_parquet_buffer)
+            assert job_runner.compute().content == expected_content
+            assert mock_ParquetFile.call_count == len(upstream_content["parquet_files"])
+            for parque_file_item in upstream_content["parquet_files"]:
+                mock_ParquetFile.assert_any_call(
+                    f"hf://datasets/{dataset}@{safe_quote(PARQUET_REVISION)}/{config}/{parque_file_item['filename']}",
+                    filesystem=get_hf_fs(app_config.common.hf_token),
+                )
+        for parquet_file_and_metadata_item in expected_content["parquet_files_and_metadata"]:
+            assert (
+                pq.read_metadata(
+                    Path(job_runner.parquet_metadata_directory)
+                    / parquet_file_and_metadata_item["parquet_metadata_subpath"]
+                )
+                == pq.ParquetFile(dummy_parquet_buffer).metadata
+            )
+
+
+def test_doesnotexist(app_config: AppConfig, get_job_runner: GetJobRunner) -> None:
+    dataset = config = "doesnotexist"
+    job_runner = get_job_runner(dataset, config, app_config)
+    with pytest.raises(CachedArtifactError):
+        job_runner.compute()
```

Comment:
btw: I don't think we really need this test (I should remove the other occurrences too)